### ISSUE TITLE
F-scores and data splits

### ISSUE THREAD

**Issue by sunguoquan1005** [e-apostolidis/PGL-SUM#1]
hello,
I reproduce the code, when I load the pretrained model, I find that the results are close to the results given in your paper. But I try the same split as "MSVA",I found that the results changed, SumMe's F1 is 51.79 when I used the pretrained model. Does this phenomenon indicate that the result has a certain relationship with the division of the dataset

_edit by mpalaourg: Format issue question and description_

**Comment by e-apostolidis** [comment]
Hello. First of all, thank you for your interest about our PGL-SUM summarization method. Concerning your question, your assumption about the existence of a relation between the utilized data and the algorithm's performance is valid. In particular, such an observation was experimentally documented in a previous work of our team (available here: http://idt.iti.gr/publications/acmmm2020_preprint.pdf) where the performance of a set of different summarization methods was evaluated on a large set of (50) randomly-created splits of data. One of the main outcomes of this work was the fact that these data splits can show varying levels of difficulty, as all tested methods exhibited similar (either low or high) levels of summarization performance on each of these splits. As a side note, our experimental evaluations showed that the performance of a random summarization can be used as an estimate of the level of difficulty, and this is why the suggested measure (Performance Over Random) involves a relevant computing step. Finally, the varying performance of a summarization method on different data is also indicated by the MSVA scores shown in Table 3 of our paper, that are different from the ones reported in the MSVA paper. And to compute these scores we did not simply run a pretrained model of the MSVA method on the test data of each of the five utilized data splits (of the SumMe and TVSum dataset), but we evaluated the MSVA method using the exact same (five sets of) training and testing data with the ones used to evaluate our method.

**Comment by sunguoquan1005** [comment]
> Hello. First of all, thank you for your interest about our PGL-SUM summarization method. Concerning your question, your assumption about the existence of a relation between the utilized data and the algorithm's performance is valid. In particular, such an observation was experimentally documented in a previous work of our team (available here: http://idt.iti.gr/publications/acmmm2020_preprint.pdf) where the performance of a set of different summarization methods was evaluated on a large set of (50) randomly-created splits of data. One of the main outcomes of this work was the fact that these data splits can show varying levels of difficulty, as all tested methods exhibited similar (either low or high) levels of summarization performance on each of these splits. As a side note, our experimental evaluations showed that the performance of a random summarization can be used as an estimate of the level of difficulty, and this is why the suggested measure (Performance Over Random) involves a relevant computing step. Finally, the varying performance of a summarization method on different data is also indicated by the MSVA scores shown in Table 3 of our paper, that are different from the ones reported in the MSVA paper. And to compute these scores we did not simply run a pretrained model of the MSVA method on the test data of each of the five utilized data splits (of the SumMe and TVSum dataset), but we evaluated the MSVA method using the exact same (five sets of) training and testing data with the ones used to evaluate our method.

Thank you for your answer,I got that

**Comment by sunguoquan1005** [comment]
I have another question, have you ever run an experiment with a fixed random number but different results. I got different results when I set the same parameter, If you experienced that, how do you deal with the problem

**Comment by sunguoquan1005** [comment]
I have another question, have you ever run an experiment with a fixed random number but  get different results. I got different results when I set the same parameter, If you experienced that, how do you deal with the problem

**Comment by e-apostolidis** [comment]
Hello. To secure reproducibility of results among different runs for a fixed random seed, we use the commands between line 26 and line 30 of solver.py (https://github.com/e-apostolidis/PGL-SUM/blob/master/model/solver.py#L26:L30). Other than that, if your network contains convolutional layers, you should pay attention to the instructions given in the documentation (https://pytorch.org/docs/stable/notes/randomness.html#avoiding-nondeterministic-algorithms) under "CUDA convolution benchmarking". If you have all these fixed, then differences among runs should be observed only after utilizing different hardware/software settings (e.g. GPU card, CUDA driver, PyTorch version, etc.).

**Comment by sunguoquan1005** [comment]
Thank you for your answer very much, and I have questions for other aspects .
First,you mentioned that the results would be affected by the hardware in your paper ,can you explain the reason for that. 
Second, did your algorithm's performance go beyond the human performance? If not,how much less can it exceed human performance. 
Third,And what is the upper bound of human performance?I found that the existing human performance is different from the human performance mentioned in the open dataset(SumMe and TVSum). If existing algorithm's performance is beyond the human performance, what can we study in video summarization field. Whether we need to develop new dataset?

**Comment by mpalaourg** [comment]
Hello, these questions go out of the scope of this repository, but I'll try to answer it as comprehensibly as I can.

> First, you mentioned that the results would be affected by the hardware in your paper ,can you explain the reason for that. 

PyTorch must have a way to compute (pseudo-) random numbers to initialize the network's weights. As noted [here](https://pytorch.org/docs/stable/notes/randomness.html), PyTorch cannot guarantee exact reproducibility across devices, and so this will affect the model's performance.

>  Second, did your algorithm's performance go beyond the human performance? If not, how much less can it exceed human performance. Third, And what is the upper bound of human performance? I found that the existing human performance is different from the human performance mentioned in the open dataset(SumMe and TVSum).

Video summarization is highly subjective, something clear even from the differences on the user summaries of each video. Numerous works talk about a way to compute the golden truth, each one with their pros and cons, and to assess the significance of the reported performance metrics (i.e. [F1 values](https://arxiv.org/pdf/1812.01969.pdf#page=11), [Correlation coefficients](https://arxiv.org/pdf/1903.11328.pdf), [Performance over random](http://idt.iti.gr/publications/acmmm2020_preprint.pdf)). 

> Whether we need to develop new dataset?

The current datasets are somehow limited to what they can give us further down the line. Of course a new dataset, with longer videos and more annotations, is always welcomed but difficult to be produced.

---
Closing the issue. If you have any more code-related questions and queries, feel free to open a new one.
